smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/providers/extension/ReactiveMessagingExtension.java

```diff
@@ -53,26 +65,13 @@
         }
     }
 
-    <T extends Emitter<?>> void processStreamEmitterInjectionPoint(@Observes ProcessInjectionPoint<?, T> pip) {
-        Channel stream = ChannelProducer.getChannelQualifier(pip.getInjectionPoint());
-        if (stream != null) {
-            emitterInjectionPoints.add(pip.getInjectionPoint());
-        }
-    }
-
-    <T extends MutinyEmitter<?>> void processStreamMutinyEmitterInjectionPoint(@Observes ProcessInjectionPoint<?, T> pip) {
-        Channel stream = ChannelProducer.getChannelQualifier(pip.getInjectionPoint());
-        if (stream != null) {
-            mutinyEmitterInjectionPoints.add(pip.getInjectionPoint());
-        }
-    }
-
-    @SuppressWarnings("deprecation")
-    <T extends io.smallrye.reactive.messaging.annotations.Emitter<?>> void processStreamLegacyEmitterInjectionPoint(
-            @Observes ProcessInjectionPoint<?, T> pip) {
+    <T> void processStreamEmitterInjectionPoint(@Observes ProcessInjectionPoint<?, T> pip) {
```

Comment:
The type parameter here isn't used anywhere, this is equivalent to `@Observes ProcessInjectionPoint<?, ?> pip`. That's generally not recommended (it will likely be slow), but if you really want the set of emitter types to be open-ended, there's no better way I guess.

Reply:
Technically we could add a marker interface to our own types and treat the spec-defined `Emitter` as a special case.

Reply:
Indeed, I added `EmitterType` marker interface. This should speed things up a bit for bean discovery.